polynote-kernel/src/main/scala/polynote/kernel/PolyKernel.scala

```diff
@@ -128,48 +137,50 @@ class PolyKernel private[kernel] (
 
   def startInterpreterFor(cellId: CellID): IO[Stream[IO, Result]] = withInterpreter(cellId) {
     (_, _, _, results) => IO.pure(results)
-  }
-
-  def runCell(id: CellID): IO[Stream[IO, Result]] = {
-    val done = ReadySignal()
-    Queue.unbounded[IO, Option[Result]].map {
-      oq =>
-          val oqSome = new EnqueueSome(oq)
-          val cellResults = Stream.eval {
-            withInterpreter(id) {
-              (notebook, cell, interp, predefResults) =>
+  } (IO.pure(Stream.empty))
+
+  def runCell(id: CellID): IO[Stream[IO, Result]] = queueCell(id).flatten
+
+  def queueCell(id: CellID): IO[IO[Stream[IO, Result]]] = {
+    taskQueue.queueTaskStream(s"Cell $id", s"Cell $id", s"Running $id") {
+      taskInfo =>
+        Queue.unbounded[IO, Option[Result]].map {
+          oq =>
+            val oqSome = new EnqueueSome(oq)
+            val run = withLaunchedInterpreter(id) {
```

Comment:
Is it guaranteed that the interpreter will be launched when this is called? I guess since this will only be called inside Subscriber... we should be ok. 

But I wonder whether we might want to have `withLaunchedInterpreter` fail if the interpreter is not launched, rather than emitting None? Since we should never really get to this point without the interpreter being launched, if it ever did happen we might want to complain about it at least.

Reply:
`withLaunchedInterpreter` is used for things that should simply do nothing if the interpreter isn't started, like completions and such. But I guess those things really want the interpreter to be started; they just can't handle the predef results. So I see your point about throwing an error there instead. I'll play with that change.